src/cplscheme/config/CouplingSchemeConfiguration.cpp

```diff
@@ -307,10 +307,29 @@ void CouplingSchemeConfiguration::xmlEndTagCallback(
   PRECICE_TRACE(tag.getFullName());
   if (tag.getNamespace() == TAG) {
     if (_config.type == VALUE_SERIAL_EXPLICIT) {
+
+      //Check the order of both participants in the explicit coupling
       if (_experimental) {
-        int maxAllowedOrder = 0; // explicit coupling schemes do not allow waveform iteration
-        checkWaveformOrderReadData(maxAllowedOrder);
+        const auto first  = _config.participants[0];
+        const auto second = _config.participants[1];
+
+        auto first_participant = _participantConfig->getParticipant(first);
+        for (auto &dataContext : first_participant->readDataContexts()) {
+          int usedOrder = dataContext.getInterpolationOrder();
+          // The first participants waveform order has to be 0 for serial explicit coupling
+          if (usedOrder != 0) {
+            PRECICE_ERROR(
+                "You configured <read-data name=\"{}\" mesh=\"{}\" waveform-order=\"{}\" />, but for the serial explicit coupling scheme only a maximum waveform-order of \"{}\" is allowed for the first participant.",
+                dataContext.getDataName(), dataContext.getMeshName(), usedOrder, 0);
+          }
+        }
+        auto second_participant = _participantConfig->getParticipant(second);
+        for (auto &dataContext : second_participant->readDataContexts()) {
+          int usedOrder = dataContext.getInterpolationOrder();
+          PRECICE_ASSERT(usedOrder >= 0);
```

Comment:
Why not also raise a proper error here? I think a variation of the one from above would be fine.

Reply:
Yes, I  have added that now.